### PR TITLE
feat: use sp1 helios logic in pallet

### PR DESCRIPTION
**Pull request by umadayal**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- [x] Feature
- [ ] Bugfix
- [x] Refactor
- [ ] Format
- [ ] Documentation
- [ ] Testing
- [ ] Other:

## Description
Refactored the Vector pallet so that it demonstrates the verification of a SP1 Helios proof. 
There is a `test_sp1_helios_proof` showing how the public values can be read from the proof file (generated from the SP1 Helios operator). Any SP1 proof in Groth16/PLONK mode can be verified with the `sp1-verifier` crate.

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -21,11 +21,13 @@ name = "avail-node"
 # Internals
 avail-base.workspace = true
 avail-core = { workspace = true, features = ["std"] }
+# avail-core.workspace = true
 kate.workspace = true
 da-runtime.workspace = true
 da-control.workspace = true
 kate-rpc.workspace = true
 testing-rpc.workspace = true
+# pallet-vector = { workspace = true, features = ["std"] }
```
Comment (ratankaliani) [inline node/Cargo.toml:30]: Remove these comments

File: pallets/vector/src/lib.rs [hunk 2]
```diff
@@ -401,72 +414,79 @@ pub mod pallet {
...
-		/// input Function input.
-		/// output Function output.
-		/// proof  Function proof.
-		/// slot  Function slot to update.
 		#[pallet::call_index(0)]
+		// Unused parameter that's hard to remove, would mess up traits
 		#[pallet::weight(weight_helper::fulfill_call::<T>(* function_id))]
 		pub fn fulfill_call(
 			origin: OriginFor<T>,
 			function_id: H256,
```
Comment (ratankaliani) [inline pallets/vector/src/lib.rs:422]: The function id is no longer needed right?

File: pallets/vector/src/lib.rs [hunk 3]
```diff
@@ -401,72 +414,79 @@ pub mod pallet {
...
+			let new_head: [u8; 8] = proof_outputs.newHead.to_le_bytes(); 
+			let new_head = u64::from_le_bytes(new_head[..8].try_into().unwrap());
+			ensure!(new_head <= head, Error::<T>::SlotBehindHead);
 
-			let config = ConfigurationStorage::<T>::get();
-			let input_hash = H256(sha2_256(input.as_slice()));
-			let output_hash = H256(sha2_256(output.as_slice()));
-			let (step_function_id, rotate_function_id) = Self::get_function_ids()?;
-			let verifier = Self::get_verifier(function_id, step_function_id, rotate_function_id)?;
+			let vkey_hash = "0x00788ce8dc2970920a3d3c072c8c07843d15f1307a53b3dd31b113c3e71c28e8";
```
Comment (ratankaliani) [inline pallets/vector/src/lib.rs:434]: Make this a constant outside of the function and rename it to `SP1_HELIOS_VKEY`.

File: pallets/vector/src/tests.rs [hunk 4]
```diff
@@ -20,14 +20,25 @@ use frame_support::{
...
+use sp1_sdk::SP1ProofWithPublicValues;
+use sp1_verifier::{Groth16Verifier, GROTH16_VK_BYTES};
 
 const TEST_SENDER_VEC: [u8; 32] =
 	hex!("d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d");
 const TEST_SENDER_ACCOUNT: AccountId32 = AccountId32::new(TEST_SENDER_VEC);
 
+// const REQUEST_ID: &str = "0x01jb00csmqe3tax204y8pe6crr";
+// const REQUEST_ID: &str = "0x01jb02hf25e3tanvqfhr2yrz7d";
+// const GROTH16_VK_BYTES: &[u8] = include_bytes!("../bn254-vk/groth16_vk.bin");
```
Comment (ratankaliani) [inline pallets/vector/src/tests.rs:39]: Remove outdated imports here

**Review by ToufeeqP — COMMENTED** [review]
The workflow results indicate duplicate lang items are still present. Can you pls address them?

**Comment by ToufeeqP** [comment]
closing in favour of #689

**Pull request closed without merging**